hazelcast/src/main/java/com/hazelcast/sql/impl/exec/Exec.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec;
+
+import com.hazelcast.sql.impl.fragment.QueryFragmentContext;
+import com.hazelcast.sql.impl.row.RowBatch;
+
+/**
+ * Basic execution stage.
+ */
+public interface Exec {
+    /**
+     * @return ID of the executor.
+     */
+    int getId();
```

Comment:
Some note on the uniqueness scope of this ID would be helpful. Is it globally unique or only in the scope of a single query/fragment? Does it stay the same between executions of the same query or valid only for a single execution?

Reply:
Added the JavaDoc. It is unique within the query operator tree. It doesn't stay between executions, because the scope of the `Exec` instance is only single query execution.